src/components/Auth/components/RegisterForm/RegisterForm.jsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import logger from 'logger.js';
+
+import { withStyles } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import Typography from '@material-ui/core/Typography';
+
+// Fields
+import { RoleField, TextField } from 'components/';
+
+// Verification
+import { FormChecker as withInputValidation } from 'hoc/';
+
+export class RegisterForm extends React.Component {
+  state = {
+    email: '',
+    password: '',
+    confirmPass: '',
+    name: '',
+    role: 'student',
```

Comment:
If we are separating the student and admin apps, should we just have the admin register on the admin app?

It's not a big deal right now, since that could require some reworking, but it's worth thinking about.

Reply:
Well, you said that there are more than 2 roles right? there is like a mentor role? or something like that?

Reply:
That's fair. I didn't really consider mentor MVP because we couldn't settle on the exact requirements for such a role

Reply:
Can we keep it on there for now? I mean i feel it is more than likely that we will be having more than one role in the future.